vector/src/main/java/im/vector/app/features/home/HomeDetailViewModel.kt

```diff
@@ -119,6 +122,14 @@ class HomeDetailViewModel @AssistedInject constructor(@Assisted initialState: Ho
             is HomeDetailAction.SwitchTab                -> handleSwitchTab(action)
             HomeDetailAction.MarkAllRoomsRead            -> handleMarkAllRoomsRead()
             is HomeDetailAction.StartCallWithPhoneNumber -> handleStartCallWithPhoneNumber(action)
+            HomeDetailAction.ViewResumed                 -> handleViewResumed()
+        }
+    }
+
+    private fun handleViewResumed() {
+        // Good place to check for layout mode
+        setState {
+            copy(layoutMode = vectorPreferences.getLayoutMode())
```

Comment:
setState only if the value is not the same? It's maybe done internally by MvRx, not sure.